### PR TITLE
fix: add grayscale colors (neutral palette) to colors.less

### PR DESCRIPTION
**Pull request by stevula**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [x] Other (about what?) - this adds Less variables for the grayscale colors mentioned in [the docs](https://ant.design/docs/spec/colors). They were previously undefined.

### 💡 Background and solution

1. The antd docs on color palettes mention a gray palette ([source](https://ant.design/docs/spec/colors#Neutral-Color-Palette)):
image

2. `ant-design-colors` also includes a gray palette for use within JS ([source](https://github.com/ant-design/ant-design-colors/blob/master/src/index.ts#L22))

3. The `@gray-*` variables are undefined in the colors.less (bundled with `antd`) which causes a compiler error if you try to import them:
image

**Solution**: added variable definitions for gray colors in colors.less

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Non-breaking change: applications can now import `@gray-1` through `@gray-10` from the Less theme. |
| 🇨🇳 Chinese | 在样式表中定义 `@gray-1` ... `@gray-10` |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed (not needed, docs already mention gray)
- [x] Demo is updated/provided or not needed (not needed)
- [x] TypeScript definition is updated/provided or not needed (no TS changes)
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/style/color/colors.less [hunk 1]
```diff
@@ -158,5 +158,17 @@
 @gold-9: color(~`colorPalette('@{gold-6}', 9) `);
 @gold-10: color(~`colorPalette('@{gold-6}', 10) `);
```
Comment (PeachScript) [inline components/style/color/colors.less:160]: Add a comment？
```less
// neutral color palette
```

File: components/style/color/colors.less [hunk 2]
```diff
@@ -158,5 +158,17 @@
...
+@gray-1: color(~`colorPalette('@{gray-6}', 1) `);
+@gray-2: color(~`colorPalette('@{gray-6}', 2) `);
+@gray-3: color(~`colorPalette('@{gray-6}', 3) `);
+@gray-4: color(~`colorPalette('@{gray-6}', 4) `);
+@gray-5: color(~`colorPalette('@{gray-6}', 5) `);
+@gray-6: @gray-base;
+@gray-7: color(~`colorPalette('@{gray-6}', 7) `);
+@gray-8: color(~`colorPalette('@{gray-6}', 8) `);
+@gray-9: color(~`colorPalette('@{gray-6}', 9) `);
+@gray-10: color(~`colorPalette('@{gray-6}', 10) `);
```
Comment (afc163) [inline components/style/color/colors.less:171]: https://github.com/ant-design/ant-design-colors/issues/39#issuecomment-651515132

Comment (afc163) [reply components/style/color/colors.less:171]: https://github.com/ant-design/ant-design-colors/issues/3#issuecomment-452158602

File: components/style/color/colors.less (hunk 1 above)
Comment (stevula) [reply components/style/color/colors.less:160]:
```suggestion
// neutral color palette
```

File: components/style/color/colors.less [hunk 3]
```diff
@@ -157,6 +157,18 @@
 @gold-8: color(~`colorPalette('@{gold-6}', 8) `);
 @gold-9: color(~`colorPalette('@{gold-6}', 9) `);
 @gold-10: color(~`colorPalette('@{gold-6}', 10) `);
+// neutral color palette
```
Comment (stevula) [inline components/style/color/colors.less:160]:
```suggestion

// neutral color palette
```

File: components/style/color/colors.less [hunk 4]
```diff
@@ -158,5 +158,18 @@
 @gold-9: color(~`colorPalette('@{gold-6}', 9) `);
 @gold-10: color(~`colorPalette('@{gold-6}', 10) `);
 
+// neutral color palette
+@gray-base: #bfbfbf;
```
Comment (stevula) [inline components/style/color/colors.less:162]: For consistency with https://github.com/ant-design/ant-design-colors/blob/abc18fd97ecd5a45f81eef70bd250319a1736fd6/src/index.ts#L22
```suggestion
@gray-base: #666666;
```

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design-colors/issues/3#issuecomment-452158602

**Comment by stevula** [comment]
Updated to use custom neutral/grayscale palette in bbd18fc:

image
https://ant.design/docs/spec/colors#Neutral-Color-Palette

**Comment by stevula** [comment]
I have made the changes. Do you know anything else or can you review again?

**Review by dforesman — APPROVED** [review]

**Review by dmarutkin — APPROVED** [review]

**Review by juanleiguarda — APPROVED** [review]

**Comment by afc163** [comment]
@zombieJ 后续统一看看，这是个遗留命名问题，感觉可以下一个大版本统一解决。

**Comment by rvinzent** [comment]
@afc163 @zombieJ with the next major version (v5.x) now released, is there any reason these changes didn't make it in? what can we do to ensure this feature gap is addressed?

**Comment by li-jia-nan** [comment]
Conflicting files, I'm sorry to close, can be re initiated if necessary.

**Pull request closed without merging**